crates/sui-rpc-api/proto/sui/rpc/v2alpha/archival_service.proto

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sui.rpc.v2alpha;
+
+import "google/protobuf/field_mask.proto";
+import "sui/rpc/v2alpha/checkpoint.proto";
+import "sui/rpc/v2alpha/executed_transaction.proto";
+import "sui/rpc/v2alpha/object.proto";
+
+// Alternitive names:
+// - HistoryService
+// - LedgerService
+// - HistoricalService
+// - HistoricalDataService
+service ArchivalService {
+  rpc GetObject(GetObjectRequest) returns (GetObjectResponse);
+  rpc BatchGetObjects(BatchGetObjectsRequest) returns (BatchGetObjectsResponse);
+
+  rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse);
+  rpc BatchGetTransactions(BatchGetTransactionsRequest) returns (BatchGetTransactionsResponse);
+
+  rpc GetCheckpoint(GetCheckpointRequest) returns (GetCheckpointResponse);
+  rpc BatchGetCheckpoints(BatchGetCheckpointsRequest) returns (BatchGetCheckpointsResponse);
+}
+
+message GetObjectRequest {
+  // Required. The `ObjectId` of the requested object.
+  optional string object_id = 1;
```

Comment:
Is this marked `optional` for schema evolution purposes? In general what's the protobuf convention for this kind of thing on input types?

Reply:
So "required" fields in terms of proto schema are not a thing anymore. They were in proto2 but there were some challenges with evolution with required fields so from proto3 and onwards you can't mark fields as required, everything is "optional" even if you don't mark it as such with the optional keyword. What the optional keyword does is ensure that generated code gives you the ability to detect field presence and distinguish between a field that is unset and one that is set with a default value. So because of this every field is marked as optional.